### ISSUE TITLE
impot LDAP users to PG

### ISSUE THREAD

**Issue by testingash** [larskanis/pg-ldap-sync#25]
Hi,
My system is ubuntu 14.04. I have installed LDAP v3, PG 9.3 and ruby 2.0
My aim is to import LDAP users to PG so I came across this post. I came across error:
ERROR:  While executing gem ... (NameError)
    uninitialized constant Gem::SafeYAML
Will appreciate any help regarding this.
Thanks.

**Comment by larskanis** [comment]
There is a problem with the ruby installation on your system. Please make sure, you updated Ubuntu-14.04:
```sh
sudo apt-get update
sudo apt-get dist-upgrade
```
I found [this bug report](https://www.mail-archive.com/search?l=ubuntu-bugs@lists.ubuntu.com&q=subject:%22%5C%5BBug+1777174%5C%5D+Re%5C%3A+2.0.0.484%5C-1ubuntu2.10+triggers+uninitialized+constant+Gem%5C%3A%5C%3ASafeYAML+on+calling+gem2.0+install%22&o=newest&f=1) with your message. You could try
```sh
ruby2.0 -r yaml -r rubygems/safe_yaml -S gem2.0 install rubygems-update
```
or
```sh
ruby2.0 -r yaml -r rubygems/safe_yaml -S gem2.0 install pg-ldap-sync
```

**Comment by testingash** [comment]
thank you for instant help. 
ruby2.0 -r yaml -r rubygems/safe_yaml -S gem2.0 install pg-ldap-sync this installed ldap-sync.

Install from Git is another option to install ldap-sync to install right?

**Comment by larskanis** [comment]
> Install from Git is another option to install ldap-sync to install right?

Right, but this seems to be an issue in rubygems and rubygems is required to install dependencies like 'net-ldap' and 'pg'.

Does pg_ldap_sync work for you with the `ruby2.0 -r yaml -r rubygems/safe_yaml -S` prefix, now?

**Comment by testingash** [comment]
No. After running command it did not given error msg but, when I tried to run test App:
cd pg-ldap-sync
  bundle install
  PATH=$PATH:/usr/lib/postgresql/10/bin/ bundle exec rake test

In bundle install command it thrown error as :
Fetching gem metadata from https://rubygems.org/.............
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Bundler could not find compatible versions for gem "ruby":
  In Gemfile:
    ruby

    bundler (~> 1.16) was resolved to 1.16.3, which depends on
      ruby (>= 1.8.7)

    pg-ldap-sync was resolved to 0.2.0, which depends on
      net-ldap (~> 0.16) was resolved to 0.16.1, which depends on
        ruby (>= 2.0.0)

    pg-ldap-sync was resolved to 0.2.0, which depends on
      pg (< 2.0, >= 0.14) was resolved to 0.18.4, which depends on
        ruby (>= 1.9.3)

    ruby-ldapserver (~> 0.3) was resolved to 0.5.3, which depends on
      ruby (>= 1.9)

Currently looking for this.
